### PR TITLE
chore(nimbus): Update generate_json_schema to use pathlib and subprocess

### LINKED ISSUE #11494: Modernize generate_json_schemas.py
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4861)

### PR DESCRIPTION
**Pull request by brennie**
Because:

- we will be adding more logic to generate_json_schema.py to generate per-model JSON schema files

this commit:

- cleans up some of the unnecessary logic; and
- modernizes the script to use pathlib, subprocess, and more types.

Fixes #11494

### CODE REVIEW

**Review by jaredlockhart — APPROVED** [review]
Oh this is the script we added as part of the pydantic 2 rewrite, eh?

Shame there isn't a more off the shelf way to handle this.

**Comment by brennie** [comment]
> Oh this is the script we added as part of the pydantic 2 rewrite, eh?
> 
> Shame there isn't a more off the shelf way to handle this.

dont worry, this is about to get even more bespoke lmao

File: schemas/generate_json_schema.py [hunk 1]
```diff
@@ -92,42 +87,51 @@ def iterate_models():
...
-    main(output)
+@click.option(
+    "--output",
+    "ts_output_path",
+    type=Path,
+    default=Path("index.d.ts"),
+    help="Output typescript file.",
+)
+def cli(**kwargs):
+    main(**kwargs)
```
Comment (mikewilli) [inline schemas/generate_json_schema.py:134]: What's the advantage of using kwargs here? My personal preference would be to be more explicit when specifying parameters to `cli()` and in calling `main()`, but is this preparing for a future capability or does it provide something else for us now?

**Review by mikewilli — COMMENTED** [review]
A couple questions. Main concern is that it appears we're no longer stripping out the `[k: string]: unknown;` lines, but maybe that's taken care of in another way?

File: schemas/generate_json_schema.py [hunk 2]
```diff
@@ -47,14 +47,9 @@ def clean_output_file(output_filename: str) -> None:
         " */\n\n",
     ]
 
-    tmp_lines = lines[:start] + lines[(end + 1) :]
-    new_lines = banner_comment_lines
-    for i, line in enumerate(tmp_lines):
-        if line.rstrip("\r\n") == "[k: string]: unknown;":
```
[the remark this answers is not in the document]
Comment (brennie) [reply schemas/generate_json_schema.py:53]: The typescript file already wasnt generating these lines.

I'm pretty sure its due to:

```
    # this prevents json2ts from adding `[k: string]: unknown;` to every interface
    if not schema.get("additionalProperties"):
        schema["additionalProperties"] = False
```

File: schemas/generate_json_schema.py (hunk 1 above)
Comment (brennie) [reply schemas/generate_json_schema.py:134]: i just removed the cli() function and we call main directly now

File: schemas/generate_json_schema.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikewilli) [reply schemas/generate_json_schema.py:53]: Ah perfect, thanks!

**Review by mikewilli — APPROVED** [review]

**Pull request merged**